ci/jenkins/test.sh

```diff
@@ -201,8 +201,13 @@ function clean_antrea {
     for antrea_yml in ${WORKDIR}/*.yml; do
         kubectl delete -f $antrea_yml --ignore-not-found=true || true
     done
-    docker images --format "{{.Repository}}:{{.Tag}}" | grep 'antrea'| xargs -r docker rmi -f || true
-    docker images | grep '<none>' | awk '{print $3}' | xargs -r docker rmi || true
+    if [[ $TESTBED_TYPE == "kind-flexible-ipam" || $TESTBED_TYPE == "kind" ]]; then
```

Comment:
Why is the cleanup logic different based on the testbed type?

Reply:
Same question, is it related to the rate limit issue?

Reply:
@antoninbas  @XinShuYang yes, no need to delete base images from the kind testbed. during run it will use images available on kind testbed no need to pull it again from docker..

Reply:
It feels like such a change does not need to be specific to Kind tests?
However, one thing to keep in mind is that we always try to build the Antrea image with the latest available version of the base images (be it ubuntu or golang) in CI. To achieve that, a `docker pull` is essentially required and it may be subject to rate limiting, regardless of whether the correct version is already present or not - but maybe we get penalized less if the image is already present (fewer requests to the docker API?).

Reply:
Since we have specified the version of the test images, this means they are unlikely to update like the latest version image. So I don't think it's necessary to remove them before each test run. Can we remove this change? @KMAnju-2021

Reply:
Yes, by this we can reduce the number of request for docker API.

Reply:
@XinShuYang , We could consider tagging the test images with latest before pushing them to the Antrea repository ?

Reply:
> @XinShuYang , We could consider tagging the test images with latest before pushing them to the Antrea repository ?

The versions of test images in conformance/networkpolicy may be updated as Kubernetes upgrades, so specifying the image versions helps us manage them more effectively. This is very important considering that we need to ensure compatibility when running CI jobs on different versions of Kubernetes clusters.

Reply:
@XinShuYang is right. Tagging everything with latest is not a good approach here. It makes it hard to track which version is actually used and obfuscates things too much. A possible approach when the same tag shows up in too many places and we want to use the same version everywhere, is to write the current version to a file. This way we only ever have to update the version / tag in one place, instead of multiple scripts / test files.

Reply:
In that case, centralising the version in a single file would be preferable for  test image version visibility, especially since we use the same test images across both regular and kind test beds. By centralising the version, we can ensure consistency across all scripts and test files, making version updates more straightforward and manageable.  Better to move ahead with centralising the versions in a separate PR.

Reply:
> In that case, centralising the version in a single file would be preferable for test image version visibility, especially since we use the same test images across both regular and kind test beds. By centralising the version, we can ensure consistency across all scripts and test files, making version updates more straightforward and manageable. Better to move ahead with centralising the versions in a separate PR.

I think it makes sense considering the single source of truth principle, but I’m okay with doing the refactor in a separate PR because it might involve refactoring both Linux and Windows jobs.

Reply:
tracked #6697